context/authentication.js

```diff
@@ -47,27 +51,15 @@ function useFirebaseAuth() {
 
   useEffect(() => firebase.auth().onAuthStateChanged(setUser), [])
 
-  const firebaseAuth = useMemo(() => {
-    return { user, promptAsync }
-  }, [user, promptAsync])
-
-  return firebaseAuth
-}
-
-export function useAuthenticationContext() {
-  return useContext(AuthenticationContext)
-}
-
-export function AuthenticationProvider({ children }) {
-  const { user, promptAsync } = useFirebaseAuth()
+  const handleLogin = useCallback(() => promptAsync(), [promptAsync])
   const handleLogout = useCallback(() => firebase.auth().signOut(), [])
   const authValue = useMemo(() => {
     return {
       user,
-      handleLogin: promptAsync,
+      handleLogin: handleLogin,
       handleLogout: handleLogout,
     }
-  }, [user, promptAsync, handleLogout])
+  }, [user, handleLogin, handleLogout])
```

Comment:
handleLogin has no reason to exist, it's just unnecessary indirection. remove it and use promptAsync directly